Assets/Scripts/PlayerController.cs

```diff
@@ -1,644 +1,544 @@
-			
-			trail.SetActive (true);
-			trail.GetComponent<Trail>().ClearSystem (true);
-			//gameObject.GetComponent<BoxCollider2D> ().enabled = true;
-			EnableShapeAndCollider();
-			inPenalty = false;
-
-	}
-
-	void RestoreMidpointMarker(){
-		foreach (GameObject mpm in GameObject.FindGameObjectsWithTag("MidpointMarker")) {
-			//mpm.SetActive (false);
-			iTween.FadeTo (mpm, 1.0f, .5f);
-		}
-	}
-	IEnumerator PlaySFXWithDelay(int whichSFX){
-		yield return new WaitForSeconds(.5f);
-		switch (whichSFX) {
-		case 1:
-			SoundManagerScript.instance.RandomizeSfx (speedUpSFX1, speedUpSFX2, speedUpSFX3);
-			break;
-		case 2:
-			SoundManagerScript.instance.RandomizeSfx (sizeUpSFX1, sizeUpSFX2);
-			break;
-		case 3:
-			SoundManagerScript.instance.RandomizeSfx (pandemoniumSFX1, pandemoniumSFX2);
-			break;
-		case 4:
-			break;
-		}
-	}
-	void ApplyPowerup(int whichPowerup, bool playSFX = true){
-		Debug.Log (whichPowerup);
-		MusicManagerScript.Instance.StartFourth ();
-		switch (whichPowerup) {
-
-		case 1:
-			speedPowerupActive = true;
-			ps.Play ();
-			speed = 20f; //was 22
-			speedPowerupTimer = 20f; 
-			if (playSFX){
-				StartCoroutine ("PlaySFXWithDelay", 1);
-			}
-			break;
-
-		case 2:
-			
-			sizePowerupActive = true;
-			gameObject.transform.localScale = new Vector3 (2f, 2f, 1f);
-			rb.mass = startMass * 2f;
-			jumpPower = startJumpPower * 1.75f;
-			sizePowerupTimer = 20f;
-			if (playSFX){
-				StartCoroutine ("PlaySFXWithDelay", 2);
-			}
-			break;
-		
-		case 3:
-			pandemoniumPowerupActive = true;
-			pandemoniumTimer = 20f;
-			if (playSFX) {
-				StartCoroutine ("PlaySFXWithDelay", 3);
-			}
-			break;
-
-		case 4:
-			int randomNum = Random.Range (1, 7);
-			//int randomNum = 6;
-			switch (randomNum) {
-			case 1:
-				ApplyPowerup (1);
-				break;
-			case 2: 
-				ApplyPowerup (2);
-				break;
-			case 3: 
-				ApplyPowerup (3);
-				break;
-			case 4: 
-				Camera.main.GetComponent<ManageWiggleScript> ().ActivateWiggle (); 
-				break;
-			case 5:
-				// broadcast to all players to activate pandemonium
-				foreach (GameObject player in GameObject.FindGameObjectsWithTag("Player")) {
-					player.GetComponent<PlayerController> ().ApplyPowerup (3, false); 
-					PlaySFXWithDelay (3);
-				}
-				foreach (GameObject mpm in GameObject.FindGameObjectsWithTag("MidpointMarker")) {
-					//mpm.SetActive (false);
-					iTween.FadeTo (mpm, 0f, .5f);
-				}
-				//Invoke ("RestoreMidpointMarker", 20f);
-
-
-				//iTween.FadeTo (midpointMarker, 0.8f, .25f);
-				break;
-			case 6:
-				foreach (GameObject player in GameObject.FindGameObjectsWithTag("Player")) {
-					player.GetComponent<PlayerController> ().ApplyPowerup (2, false); 
-					PlaySFXWithDelay (2);
-				}
-				break;
-			}
-			break;
-		
-		}
-	
-	}
-}
+        // Get collider for chosen shape
+        shapeCollider.enabled = true;
+
+        // Push play event
+        string shapePlayEvent = playerShape + "Plays";

```

Comment:
Nice, I see how this kind of consolidating is making stuff like this simpler and less 'if / else' or 'switch' block-y